app/models/category_list.rb

```diff
@@ -63,9 +63,13 @@ def find_relevant_topics
 
     @all_topics = Topic.where(id: category_featured_topics.map(&:topic_id))
     @all_topics = @all_topics.includes(:last_poster) if @options[:include_topics]
+
+    @dismissed_topic_users_lookup = DismissedTopicUser.lookup_for(@guardian.current_user, @all_topics) if @guardian.current_user
+
     @all_topics.each do |t|
       # hint for the serializer
       t.include_last_poster = true if @options[:include_topics]
+      t.dismissed = @guardian.current_user && @dismissed_topic_users_lookup.include?(t.id)
```

Comment:
```suggestion
t.dismissed = dismissed_topic?(topic)

def dismissed_topic?(topic)
  if @guardian.current_user
    @dismissed_topic_users_lookup ||= DismissedTopicUser.lookup_for(@guardian.current_user, @all_topics)
    @dismissed_topic_users_lookup.include?(topic.id)
  else
    false
  end
end
```

What do you think about simplifying the method like so? That way we don't have to keep checking for `@guardian.current_user`